pkgs/by-name/ma/materialize/package.nix

```diff
@@ -168,6 +170,13 @@ rustPlatform.buildRustPackage rec {
     install --mode=444 -D ./misc/dist/materialized.service $out/etc/systemd/system/materialized.service
   '';
 
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgram = "${placeholder "out"}/bin/clusterd";
```

Comment:
I think we can add a meta mainProgram as well.

Reply:
Well, no matter if we define `meta.mainProgram` or not, `versionCheckProgramHook` will ignore it and solely try `pname`.
Also, this package ships two executables: `clusterd` and `environmentd`. I am not sure which to put in `meta.mainProgram`.

What do you think ? Should we still put one ?

Reply:
`environmentd` is the main program of `materialize`

Reply:
Thanks ! I updated the package and used `environmentd` in `versionCheckHook` too.